lib/broadway/batcher.ex

```diff
@@ -15,18 +15,37 @@ defmodule Broadway.Batcher do
   def init(args) do
     Process.put(@all_batches, %{})
     batcher = args[:batcher]
+    partition_by = args[:partition_by]
 
     state = %{
       batcher: batcher,
       batch_size: args[:batch_size],
       batch_timeout: args[:batch_timeout]
     }
 
+    n_consumers = args[:n_consumers]
+
+    dispatcher_args =
+      case partition_by do
+        nil ->
+          []
+
+        func ->
+          hash_func = fn {[msg | _], _info} = payload ->
```

Comment:
Ideally, we should use put/use the information from `batch_info` instead of the first message.